### DISCUSSION TITLE
Excessive time duration on a vDF creation

### DISCUSSION THREAD

**Discussion by corentin2l** [vertica/VerticaPy#1234] [Potential bug] [answered]
Hello,

We detected a different behaviour of verticapy between the v0.12.0 and the v1.0.3.

When we construct a vDF on the V0.12.0, each request sent have the parameter "limit 0".
So, each sent request doesn't have to wait for an answer but it just tests if the request is correct.

On the v1.0.3, the parameter seems to have been removed. So, each request waits for an answer.

These simple requests only need few seconds to be executed on the V0.12.0, but a long time on the V1.0.3.

base = vpy.vDataFrame(input_relation = "schema.base", )
base = base.select(['"COLUMNS1"', '"COLUMNS2"', '"COLUMNS3"'])
base = base.search(PERIODE_FILTER)

Here are the the durations of the query, on both versions :
on v0.12.0
![image](https://github.com/vertica/VerticaPy/assets/43996984/b12a2070-e3ae-407b-9c4d-fa0c9bd39ea7)

on v1.0.3
![image](https://github.com/vertica/VerticaPy/assets/43996984/f0f837c2-e396-4622-a0d8-e4816c340649)

How can we create a vDF like the V0.12.0, with the parameter limit 0 or equivalent on V1.0.3 ?

Regards,

Corentin

**Comment by oualib** [comment]
Thanks Corentin for the update. An issue has been created: https://github.com/vertica/VerticaPy/issues/1235
We will try to investigate to see why you're facing such problem.

**Reply by oualib** [reply, marked as answer]
Solved here: https://github.com/vertica/VerticaPy/pull/1254
Indeed, a LIMIT 0 has to be added in the get_datatypes. It solved the overall problem. Thanks for reporting this.
